### ISSUE TITLE
Animation glitching on Flutter web.

### ISSUE THREAD

**Issue by Imgkl** [rive-app/rive-flutter#290]
### Description
On Mobile, The animation is rendered in a weird state, the same works on Desktop browser

### Steps To Reproduce
1. I downloaded this asset after removing the background from Rive editor (https://rive.app/community/2568-5220-flux-capacitor/)

2. Added Rive asset to the widget
` RiveAnimation.asset('assets/flux.riv'))`

4. On render, the animation gets glitched

#### Source `.riv`/`.rev` file
https://rive.app/community/2568-5220-flux-capacitor

### Expected behavior
The animation should be rendered properly

#### Screenshots
<img src="https://user-images.githubusercontent.com/25414681/216726814-b6b20cc7-4b42-4d03-bced-100c360053bc.jpeg"  height="600">
<img src="https://user-images.githubusercontent.com/25414681/216726824-ad4fbd72-88ee-43d0-93cf-238adf0750f4.jpeg"  height="600">

![Screenshot](url)

### Device & Versions (please complete the following information)
- Device: iPhone 11 
- OS: iOS 16.3
- Flutter Version: 
```
Flutter 3.3.10 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 135454af32 (7 weeks ago) • 2022-12-15 07:36:55 -0800
Engine • revision 3316dd8728
Tools • Dart 2.18.6 • DevTools 2.15.0
```

**Comment by HayesGordon** [comment]
Hey @Imgkl

Wow! Awesome animation!

I just tested on my machine (M1):
```
Flutter 3.8.0-1.0.pre.11 • channel master • https://github.com/flutter/flutter.git
Framework • revision 254a796bc3 (5 days ago) • 2023-02-01 09:19:48 -0800
Engine • revision c08a286d60
Tools • Dart 3.0.0 (build 3.0.0-189.0.dev) • DevTools 2.21.1
```

And it seemed to render fine (android, iOS, and macOS). So it may be that a newer version of Flutter could solve this for you.

I'll test in a bit to see if I can replicate the issue on the Flutter stable branch.

Which version of the Rive runtime are you using? I tested on the latest v0.10.1

**Comment by Imgkl** [comment]
@HayesGordon Thanks for reverting back.

The credits for the animation goes to JcToon. I'm just using it as preloader for my app.

I'll update my Flutter SDK to latest and will revert to you. 

Also FYI, I'm building this app on Flutter web, Also I noticed a couple of other animations looks odd on real device (I'm using iOS), but on desktop browser, in responsive mode, they looks fine.

**Comment by HayesGordon** [comment]
@Imgkl Okay, I see. So this is a web build, and on the mobile browsers you are experiencing these issues? Did you only test on an iOS device? And which browser are you using?

**Comment by Imgkl** [comment]
Updated flutter to version 3.7 and same result.

**Comment by SolweyWes** [comment]
Similar issue, not consistent between files - https://rive.app/community/3827-8012-onoff-button/ only stays in 'Pressed State,' but two other Rive samples (Moon Scan and Star Rating) seem to work fine.


@Imgkl what controller are you using?

**Comment by Imgkl** [comment]
@SolweyWes The above animation I posted is one-shot. So I'm not using any controllers for that.

**Comment by HayesGordon** [comment]
Hi @Imgkl, I was able to confirm the issue you reported. It appears to be a problem with Flutter/Skia on the different platforms.

We're about to release an update to the Flutter runtime that just so happens to solve this particular issue. Can you confirm on your mobile devices if the following renders correctly: https://incomparable-dragon-7d9a6a.netlify.app/#/

**Comment by HayesGordon** [comment]
Hi @SolweyWes can you share a reproducible example of the issue you're running in to and I'll take a look. There might be something going on with how you are using the controllers

**Comment by Imgkl** [comment]
@HayesGordon the animation in the above netlify works completely fine. No glitches of any sort. Although the text is kinda illegible but the overall animation is definitely a step up from what we have currently.

![IMG_9A12C99FDAF2-1](https://user-images.githubusercontent.com/25414681/218711692-c539111e-bf61-4bea-9b06-82cfa09f79ee.jpeg)

So what changes should I make to fix this in my app?

**Comment by HayesGordon** [comment]
@Imgkl see here for additional details: https://github.com/flutter/flutter/issues/97237#issuecomment-1035368982

By default Flutter uses the HTML renderer for Flutter web on mobile and the differences between HTML and Canvaskit are outside of our control. You can build your Web project to be only CanvasKit:
```
flutter build web --web-renderer canvaskit
```

Here is a sample and this should not have any of the problems you mentioned: https://sensational-panda-5df33f.netlify.app/#/

**Comment by HayesGordon** [comment]
This is now resolved in v0.10.2 of the runtime.

Not the differences between CanvasKit and HTML renderer, but the main thing where the opacity is not working correctly.

In general this particular animation should run a lot smoother as well.

**Comment by Imgkl** [comment]
Works completely fine after updating the runtime. Thanks @HayesGordon
